keyboards/primekb/prime_e/keymaps/madhatter/keymap.c

```diff
@@ -0,0 +1,165 @@
+    KC_LCTL, KC_LALT,          LOWER,  KC_LCMD,    LT(_RAISE, KC_SPC),  KC_LALT,          KC_RGUI, KC_RCTL
+  ),
+
+  [_LOWER] = LAYOUT(
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  KC_DEL,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_SCLN, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_LBRC, KC_RBRC, KC_BSLS, KC_TRNS,
+    KC_TRNS, KC_TRNS,                     LOWER, KC_TRNS,   KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS
+  ),
+
+  [_RAISE] = LAYOUT(
+    KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,   KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_MINS,  KC_EQL, KC_LCBR, KC_RCBR, KC_PIPE, KC_GRV,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_UNDS, KC_PLUS, KC_LBRC, KC_RBRC, KC_BSLS, KC_TILD,
+    KC_TRNS, KC_TRNS,                     LOWER, KC_TRNS,   KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS
+  ),
+
+  [_ADJUST] = LAYOUT(
+      RESET, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,     KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+    KC_CAPS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    AG_TOGG, KC_TRNS,                     LOWER, KC_TRNS,   KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS
+  )
+};
+
+void matrix_init_user(void) {
+  // set CapsLock LED to output and low
+  setPinOutput(CAPS_LED_PIN);
+  writePinLow(CAPS_LED_PIN);
+  // set NumLock LED to output and low
+  setPinOutput(NUM_LED_PIN);
+  writePinLow(NUM_LED_PIN);
+  // set ScrollLock LED to output and low
+  setPinOutput(SCROLL_LED_PIN);
+  writePinLow(SCROLL_LED_PIN);
+}
+
+void matrix_scan_user(void) {
+    if (CAPS == 1) {
+        // Blink the first led when capslock is active
+        if (BLINK_TIMER >= BLINK_DURATION) {
+            togg_indicator(&CAPS_LED_STATE, CAPS_LED_PIN);
+            BLINK_TIMER = 0;
+        }
+        BLINK_TIMER++;
+    }
+}
+
+//function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    if (layer_state_cmp(state, 1)) {
+        writePinHigh(CAPS_LED_PIN);
+    } else {
+        writePinLow(CAPS_LED_PIN);
+    }
+    if (layer_state_cmp(state, 2)) {
+        writePinHigh(NUM_LED_PIN);
+    } else {
+        writePinLow(NUM_LED_PIN);
+    }
+    if (layer_state_cmp(state, 3)) {
+        writePinHigh(SCROLL_LED_PIN);
+    } else {
+        writePinLow(SCROLL_LED_PIN);
+    }
+    return state;
+}
+
+void led_set_user(uint8_t usb_led) {
+    if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
+        CAPS = 1;
+    }
+    else {
+        CAPS = 0;
+    }
+}
```

Comment:
I mean, you really just need: 
```suggestion
void led_set_user(uint8_t usb_led) {
    CAPS = IS_LED_ON(usb_led, USB_LED_CAPS_LOCK);
}
```

Or better: 
```suggestion
bool led_update_user(led_t led_state) {
    CAPS = led_state.caps_lock;
    return true;
}
```